### PR TITLE
Reply to comments feature #95

### PR DESCRIPTION
**Pull request by Daniel194**

### CODE REVIEW

**Comment by georgebejan** [comment]
A `Reply` is still a `Comment`. Not sure why do we need a separate entity to represent replies.
I would prefer to just update the `Comment` entity to support some sort of tree structure (since in theory we could have multiple levels of replies). This way each reply is still a `Comment` but it will have a `parentId` or something similar.

**Comment by RaduCStefanescu** [comment]
> A `Reply` is still a `Comment`. Not sure why do we need a separate entity to represent replies.
> I would prefer to just update the `Comment` entity to support some sort of tree structure (since in theory we could have multiple levels of replies). This way each reply is still a `Comment` but it will have a `parentId` or something similar.

I second what @georgebejan is saying here. I don't see the point of adding a new resource in order to capture comment replies. We should extend the comment resource and use it for replies as well.

**Comment by Daniel194** [comment]
I changed `Comment`  entity to be used as reply also. 

Two new REST services were added, one to create a reply and one to read all the replies for a specific comment. For update and delete a reply we could use the REST services which already exist for comments.

File: src/main/java/com/code4ro/legalconsultation/service/impl/CommentServiceImpl.java [hunk 1]
```diff
@@ -69,6 +69,22 @@ public CommentDto create(UUID nodeId, final CommentDto commentDto) {
...
+    @Transactional
+    @Override
+    public CommentDto createReply(UUID parentId, CommentDto commentDto) {
+        Comment parent = commentRepository.findById(parentId).orElseThrow(EntityNotFoundException::new);
+        ApplicationUser currentUser = currentUserService.getCurrentUser();
+
+        Comment reply = mapperService.map(commentDto, Comment.class);
+        reply.setParent(parent);
+        reply.setOwner(currentUser);
+        reply.setLastEditDateTime(new Date());
```
Comment (RaduCStefanescu) [inline src/main/java/com/code4ro/legalconsultation/service/impl/CommentServiceImpl.java]: reply.setLastEditDateTime(LocalDate.now())

Comment (RaduCStefanescu) [inline src/main/java/com/code4ro/legalconsultation/service/impl/CommentServiceImpl.java]: `reply.setLastEditDateTime(LocalDate.now());` ?

**Pull request merged**